foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClient.java

```diff
@@ -110,65 +111,128 @@ public static OpenAiClient withCustomDestination(@Nonnull final Destination dest
+    prompt.messages().forEach(message -> request.addMessagesItem(message.createDTO()));
+    request.stop(CreateChatCompletionRequestAllOfStop.create(prompt.stop()));
+
+    request.temperature(config.temperature());
+    request.topP(config.topP());
+    request.stream(config.stream());
+    request.maxTokens(config.maxTokens());
+    request.maxCompletionTokens(config.maxCompletionTokens());
+    request.presencePenalty(config.presencePenalty());
+    request.frequencyPenalty(config.frequencyPenalty());
+    request.logitBias(config.logitBias());
+    request.user(config.user());
+    request.logprobs(config.logprobs());
+    request.topLogprobs(config.topLogprobs());
+    request.n(config.n());
+    request.parallelToolCalls(config.parallelToolCalls());
+    request.seed(config.seed());
+    request.streamOptions(config.streamOptions());
+    request.responseFormat(config.responseFormat());
+    request.tools(config.tools());
+    request.toolChoice(config.toolChoice());
+    request.functionCall(null);
+    request.functions(null);
+    return request;
+  }
+
   /**
-   * Add a system prompt before user prompts.
+   * Use this method to set a system prompt that should be used across multiple chat completions
+   * with basic string prompts {@link #streamChatCompletionDeltas(OpenAiChatCompletionPrompt)}.
+   *
+   * <p>Note: The system prompt will be ignored on chat completions invoked with
+   * OpenAiChatCompletionPrompt.
    *
    * @param systemPrompt the system prompt
    * @return the client
    */
   @Nonnull
   public OpenAiClient withSystemPrompt(@Nonnull final String systemPrompt) {
-    this.systemPrompt = systemPrompt;
+    this.systemPrompt = OpenAiMessage.system(systemPrompt);
     return this;
   }
 
   /**
-   * Generate a completion for the given user prompt.
+   * Generate a completion for the given string prompt as user.
    *
    * @param prompt a text message.
    * @return the completion output
    * @throws OpenAiClientException if the request fails
    */
   @Nonnull
-  public CreateChatCompletionResponse chatCompletion(@Nonnull final String prompt)
+  public OpenAiChatCompletionOutput chatCompletion(@Nonnull final String prompt)
       throws OpenAiClientException {
-    final CreateChatCompletionRequest parameters = new CreateChatCompletionRequest();
+    final var userPrompt = OpenAiMessage.user(prompt);
 
-    if (systemPrompt != null) {
-      parameters.addMessagesItem(
-          new ChatCompletionRequestSystemMessage()
-              .role(ChatCompletionRequestSystemMessage.RoleEnum.SYSTEM)
-              .content(ChatCompletionRequestSystemMessageContent.create(systemPrompt)));
-    }
-    parameters
-        .addMessagesItem(
-            new ChatCompletionRequestUserMessage()
-                .role(ChatCompletionRequestUserMessage.RoleEnum.USER)
-                .content(ChatCompletionRequestUserMessageContent.create(prompt)))
-        .functions(null)
-        .tools(null)
-        .parallelToolCalls(null);
```

Comment:
(Note)

This branch is setting `null` implicitly via passing `OpenAiChatCompletionConfig` fields, which are initialized as `null` in case of `List` value type.